project-management/src/main/java/life/qbic/projectmanagement/application/measurement/MeasurementService.java

```diff
@@ -70,9 +70,23 @@ private static Optional<ProteomicsMeasurementMetadata> merge(
     }
     List<SampleCode> associatedSamples = measurementMetadataList.stream().map(
         ProteomicsMeasurementMetadata::sampleCodes).flatMap(Collection::stream).toList();
+    List<Labeling> labels = measurementMetadataList.stream().map(
+        metadata -> {
+          Labeling labeling;
+          try {
+            labeling = metadata.labeling().iterator().next();
+          } catch (Exception e) {
+            log.warn(
+                "Proteomics label: No label information found for pooling %s".formatted(
+                    metadata.sampleCodes().iterator().next().code()));
+            labeling = new Labeling(metadata.sampleCodes().iterator().next().code(), "", "");
```

Comment:
the iterator of the sample codes and the iterator of the labeling do not return their elements in a specific order. According to the method signature we get a Collection here which does not provide ordered elements (might if it is actually a list but this is not guaranteed by the method signature). If you assume that only one sample and one labelling information is available, please check the assumption separately and fetch the information. 
It might be a good idea to extract a method for the lambda here since it is very complex to read.